x/staking/keeper/liquid_stake_test.go

```diff
@@ -0,0 +1,514 @@
+			newShares:           sdk.NewDec(110),
+			expectedExceeds:     false,
+		},
+		{
+			// Validator Shares: 100, Factor: 2, Current Shares: 90 => 200 Max Shares, Capacity: 110
+			// New Shares: 111 - above cap
+			name:                "factor 2 - above cap",
+			validatorShares:     sdk.NewDec(100),
+			validatorBondFactor: sdk.NewDec(2),
+			currentLiquidShares: sdk.NewDec(90),
+			newShares:           sdk.NewDec(111),
+			expectedExceeds:     true,
+		},
+		{
+			// Validator Shares: 100, Factor: 100, Current Shares: 90 => 10000 Max Shares, Capacity: 9910
+			// New Shares: 100 - below cap
+			name:                "factor 100 - below cap",
+			validatorShares:     sdk.NewDec(100),
+			validatorBondFactor: sdk.NewDec(100),
+			currentLiquidShares: sdk.NewDec(90),
+			newShares:           sdk.NewDec(100),
+			expectedExceeds:     false,
+		},
+		{
+			// Validator Shares: 100, Factor: 100, Current Shares: 90 => 10000 Max Shares, Capacity: 9910
+			// New Shares: 9910 - at cap
+			name:                "factor 100 - at cap",
+			validatorShares:     sdk.NewDec(100),
+			validatorBondFactor: sdk.NewDec(100),
+			currentLiquidShares: sdk.NewDec(90),
+			newShares:           sdk.NewDec(9910),
+			expectedExceeds:     false,
+		},
+		{
+			// Validator Shares: 100, Factor: 100, Current Shares: 90 => 10000 Max Shares, Capacity: 9910
+			// New Shares: 9911 - above cap
+			name:                "factor 100 - above cap",
+			validatorShares:     sdk.NewDec(100),
+			validatorBondFactor: sdk.NewDec(100),
+			currentLiquidShares: sdk.NewDec(90),
+			newShares:           sdk.NewDec(9911),
+			expectedExceeds:     true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Update the validator bond factor
+			params := app.StakingKeeper.GetParams(ctx)
+			params.ValidatorBondFactor = tc.validatorBondFactor
+			app.StakingKeeper.SetParams(ctx, params)
+
+			// Create a validator with designated self-bond shares
+			validator := types.Validator{
+				TotalLiquidShares:        tc.currentLiquidShares,
+				TotalValidatorBondShares: tc.validatorShares,
+			}
+
+			// Check whether the cap is exceeded
+			actualExceeds := app.StakingKeeper.CheckExceedsValidatorBondCap(ctx, validator, tc.newShares)
+			require.Equal(t, tc.expectedExceeds, actualExceeds, tc.name)
+		})
+	}
+}
+
+// Tests SafelyIncreaseValidatorTotalLiquidShares
+func TestSafelyIncreaseValidatorTotalLiquidShares(t *testing.T) {
+	_, app, ctx := createTestInput(t)
+
+	initialLiquidShares := sdk.NewDec(0)
+	validatorBondShares := sdk.NewDec(10)
+	increaseAmount := sdk.NewDec(20)
+
+	// Create a validator with designated self-bond shares
+	privKey := secp256k1.GenPrivKey()
+	pubKey := privKey.PubKey()
+	valAddress := sdk.ValAddress(pubKey.Address())
+
+	initialValidator := types.Validator{
+		OperatorAddress:          valAddress.String(),
+		TotalLiquidShares:        initialLiquidShares,
+		TotalValidatorBondShares: validatorBondShares,
```

Comment:
I see in these tests we set the ValidatorBondShares directly in state. The other PR tests the process of delegating -> marking delegation as ValidatorBond ?

Reply:
yeah this is just testing [this](https://github.com/iqlusioninc/liquidity-staking-module/pull/61/files/744377e109d1430f13b3572bb6cf3a686f4d6df6#diff-fb1fb56df25f422dcd246c4517169542f2a04e040d8e39d261ee8ee1477bee46R83) function. 

All that matters is that there's a validator in the store with liquid and bonded shares - no reason to complicate/bloat this test!